kernel/mempool.c

```diff
@@ -46,9 +62,24 @@ void k_free(void *ptr)
 K_HEAP_DEFINE(_system_heap, CONFIG_HEAP_MEM_POOL_SIZE);
 #define _SYSTEM_HEAP (&_system_heap)
 
-void *k_malloc(size_t size)
+void *k_aligned_alloc(size_t align, size_t size)
```

Comment:
Since we're adding a new API, should we consider having the align parameter be an exponent like IIRC Linux does? that would reduce the number of checks done

Reply:
the underlying heap implementation as well as the semantic / nominative reference (posix_memalign / aligned_alloc) both specify alignment in bytes, so I would just as soon leave it.

Reply:
Also, the "popcount" check was removed since it was already an assert in sys_heap_alloc_aligned()